### PR TITLE
Metadata ui

### PR DESCRIPTION
**Pull request by kathryn1995**

### CODE REVIEW

File: ui/src/components/UpdateSamplesManager.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      return;
+    },
+    onError() {
+      setSheet(null);
+      setIsCommitted(false);
+      return;
+    },
+  });
+
+  return isAdmin ? (
```
Comment (garethpeat) [inline ui/src/components/UpdateSamplesManager.js:53]: Perhaps overusing ternaries here. The rendering code logic might be more readable with explicit `if...else` branched `return` statements. Maybe we should discuss this.

Comment (garethpeat) [reply ui/src/components/UpdateSamplesManager.js:53]: Some thoughts:

*page structure*
- could conditionally show the dropzone or the filename (but always one or the other)
- added/removed/same/changed should be shown/hidden if diff calculated
- should perhaps always show the Commit/Cancel buttons (but conditionally disabled)

*modals*
- should not directly affect the above being rendered on the update page (just indirectly via state changes)
- should either redirect or set state, then close
- might be helpful to have named handlers, such as:
  - `cancel`, which redirects to `/`
  - `clean`, which doesn't redirect, but sets sheet, isCommittable and isCommitted

File: ui/src/components/FailedDiffModal.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 2, 3),
+    color: "white",
+  },
+}));
+
+function FailedDiffModal({ showModal }) {
```
Comment (garethpeat) [inline ui/src/components/FailedDiffModal.js:29]: It looks like `FailedDiffModal` could be removed?

File: ui/src/components/CommitMetadataModal.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 2, 3),
+    color: "white",
+  },
+}));
+
+function CommitModal({ showModal, setIsCommitted, isCommitted }) {
```
Comment (garethpeat) [inline ui/src/components/CommitMetadataModal.js:29]: It looks like `CommitModal` could be removed?

File: ui/src/components/UpdateSamplesManager.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        Clear
+      </Button>
+      {/* MODALS */}
+      <GenericDialog
+        showModal={isCommitted}
+        title={"Updated"}
+        text={
+          "Your commit was successful! Click okay to return to the home page or cancel to upload another spreadsheet."
+        }
+        onOk={() => {
```
Comment (garethpeat) [inline ui/src/components/UpdateSamplesManager.js:104]: Is `onOk={routeHome}` sufficient here?

File: ui/src/components/UpdateSamplesManager.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+          routeHome();
+          refreshPage();
+        }}
+        onCancel={() => {
```
Comment (garethpeat) [inline ui/src/components/UpdateSamplesManager.js:108]: Is onCancel={handleClose} sufficient here?

File: ui/src/components/UpdateSamplesManager.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }
+`;
+
+const UpdateSamplesManager = () => {
+  const { isAdmin } = useUser();
+  const [sheet, setSheet] = useState(null);
+  const [isCommittable, setIsCommittable] = useState(false);
+  const [isCommitted, setIsCommitted] = useState(false);
+  const [commitFailed, setCommitFailed] = useState(false);
```
Comment (garethpeat) [inline ui/src/components/UpdateSamplesManager.js:40]: The looks good; the state is much clearer now.

**Review by garethpeat — COMMENTED** [review]
Nice. `UpdateSamplesManager` logic now much clearer.

File: ui/src/components/UpdateSamplesDiff.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        laneId
+      }
+      missingInstitutions
+    }
+  }
+`;
+
+const UpdateSamplesDiff = ({ sheet, setIsCommittable }) => {
+  const classes = useStyles();
+  const [diffPass, setDiffPass] = useState(null);
```
[the remark this answers is not in the document]
Comment (kathryn1995) [reply ui/src/components/UpdateSamplesDiff.js]: Removed diffPass

File: ui/src/components/UpdateSamplesDiff.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+  const { loading, error, data } = useQuery(DIFF_QUERY, {
```
[the remark this answers is not in the document]
Comment (kathryn1995) [reply ui/src/components/UpdateSamplesDiff.js:58]: Removed this

File: ui/src/components/UpdateSamplesManager.js (hunk 5 above)
Comment (kathryn1995) [reply ui/src/components/UpdateSamplesManager.js]: Changed query so can resubmit new spreadsheet and it will print correct stats

**Pull request merged**